corehq/apps/registration/validation.py

```diff
@@ -40,3 +51,12 @@ def validate_profile(self, new_profile_name):
         profile_validator = ProfileValidator(self.domain, self.upload_user, True, self.profiles_by_name())
         spec = {'user_profile': new_profile_name}
         return profile_validator.validate_spec(spec)
+
+    def validate_email(self, email, is_post):
+        if is_post:
```

Comment:
Why `is_post` flag is required here? What can happen if we don't have this parameter?

Reply:
The `is_post` is needed in preparation for it being used for the Create Invitation / Update Web User API.  When we're creating a new invitation, we need to check that the email isn’t already associated with an existing user or invitation. But if we’re updating it, we already know the user or invitation exists, so running the same check would always fail.